neo/Consensus/ConsensusContext.cs

```diff
@@ -253,5 +256,34 @@ public bool VerifyRequest()
             if (tx_gen?.Outputs.Sum(p => p.Value) != amount_netfee) return false;
             return true;
         }
+
+        public uint GetSnapshotTimestamp()
+        {
+            return Snapshot.GetHeader(PrevHash).Timestamp;
+        }
+
+        public uint GetLimitTimestamp()
+        {
+            return DateTime.UtcNow.AddMinutes(10).ToTimestamp();
+        }
+
+        public uint GetCurrentTimestamp()
+        {
+            return Math.Max(DateTime.UtcNow.ToTimestamp(), Snapshot.GetHeader(PrevHash).Timestamp + 1);
```

Comment:
`GetUtcNow()` instead of `DateTime.UtcNow`?

Reply:
Yeah, because GetUtcNow will become abstract when IConsensusContext is created, so we will be able to manipulate the timings and nonce to generate specific blocks on testing. The whole plan is on my mind, but we need to move step by step, so everyone is fully aware...

Reply:
Smart guy !! just got your point Shargon... lets use GetUtcNow and GetSnapshotTimestamp, so the rest wont need to become virtual ;)